core/src/main/java/io/fabric8/maven/core/util/SpringBootUtil.java

```diff
@@ -40,16 +38,39 @@
     private static final transient Logger LOG = LoggerFactory.getLogger(SpringBootUtil.class);
 
     /**
+     * This method computes the active profiles and delegate the call to the getApplicationProperties method
+     *
+     * @param project        - the {@link MavenProject}
+     * @param activeProfiles - the comma separated String of profiles
+     * @return properties - the merged properties of all profiles
+     */
+    public static Properties getApplicationProperties(MavenProject project, String activeProfiles) {
+        return getApplicationProperties(project, getActiveProfiles(activeProfiles));
+    }
+
+    /**
      * Returns the spring boot configuration (supports `application.properties` and `application.yml`)
      * or an empty properties object if not found
      */
-    public static Properties getSpringBootApplicationProperties(MavenProject project) {
+    public static Properties getApplicationProperties(MavenProject project, List<String> activeProfiles) {
         URLClassLoader compileClassLoader = MavenUtil.getCompileClassLoader(project);
-        URL ymlResource = compileClassLoader.findResource("application.yml");
-        URL propertiesResource = compileClassLoader.findResource("application.properties");
+        URL ymlResource = getResourceFromClasspath(compileClassLoader, "application.yml");
+        URL propertiesResource = getResourceFromClasspath(compileClassLoader, "application.properties");
 
-        Properties props = getPropertiesFromYamlResource(ymlResource);
+        Properties props = getPropertiesFromYamlResource(ymlResource, activeProfiles);
         props.putAll(getPropertiesResource(propertiesResource));
+
+        //If the profiles are available load the profile resources as well
+        if (activeProfiles != null) {
+            for (String profile : activeProfiles) {
+                ymlResource = getResourceFromClasspath(compileClassLoader, "application-" + profile + ".yml");
+                Properties props2 = getPropertiesFromYamlResource(ymlResource, activeProfiles);
+                propertiesResource = getResourceFromClasspath(compileClassLoader,
+                        "application-" + profile + ".properties");
+                props2.putAll(getPropertiesResource(propertiesResource));
+                props.putAll(props2);
```

Comment:
You could put all of this logic into a method (from line 66 to line 71) and then call it for the case above the if, too (which has the same logic but without a profile name).

I.e. I would add a `addApplicationProperties(props, compileClassLoader, profile)` with `profile == null` means no profile.